velox/exec/SortBuffer.cpp

```diff
@@ -263,8 +266,14 @@ void SortBuffer::ensureOutputFits() {
     return;
   }
 
+  const vector_size_t numRows =
+      output_ == nullptr ? batchSize : std::max(0, batchSize - output_->size());
+  if (numRows == 0) {
+    return;
+  }
+
   const uint64_t outputBufferSizeToReserve =
-      estimatedOutputRowSize_.value() * 1.2;
+      estimatedOutputRowSize_.value() * numRows;
```

Comment:
If previously we had 1.2 as factor, why not to use `estimatedOutputRowSize_.value() * numRow * 1.2`?

Reply:
I don't think we need the factor, the estimate size is for row format, but the output is column format, the size of latter is smaller than former. And I test the `TEST_F(RowContainerTest, estimateRowSize)`, compare the data size with rowVector->estimateFlatSize(), row format is 64 while column format is 45 when the number of rows is 200,000 and data type is (BIGINT, VARCHAR).

Reply:
Thank for the explanation. Though I think for this kind of estimation we may take the worst case into account? Say if the batch only has 100 rows or even less. Is the calculation still safe in the case?

All based on the assumption that when the calculation fails, OOM will definitely take place which is unexpected.

Reply:
Yes, addressed.

Reply:
If we need the 1.2 factor, why don't we put it in higher level like estimatedOutputRowSize_'s calculation to make sure all estimation is the over estimated? Also we need to document it as a guideline.